tests/service-worker/test.js

```diff
@@ -1,36 +1,135 @@
+    wpSwCache.storage.clear().then(function() {
+      self.caches.delete(wpSwCache.cacheName).then(function(){
+        done();
+      });
+    });
   });
 
   it('This is a stub test', function() {
-    assert.strictEqual(true, true);
+    assert.isTrue(true);
   });
 
-  it('URLs which should be added to cache actually end up in cache and storage', function() {
-    // Simulates a basic install
-    assert.strictEqual(true, true);
+  it('URLs which should be added to cache actually end up in cache and storage', function(done) {
+    this.timeout(10000);
+
+    // Simulates a basic install -- check cache and localforage
+    wpSwCache.update().then(function() {
+      return self.caches.open(wpSwCache.cacheName).then(function(cache) {
+        return cache.keys().then(function(keys) {
+          assert.isTrue(keys.length == 2);
+
+          var cacheMatches = 0;
+          var storageMatches = 0;
+          return Promise.all(Object.keys(wpSwCache.urls).map(function(key) {
+            return cache.match(key).then(function(result) {
+              var isGood = (result && result.url === key);
+              assert.isTrue(isGood);
+              if(isGood) {
+                cacheMatches++;
+              }
+            })/*().then(function() {
+              return wpSwCache.storage.getItem(key).then(function(result) {
+                var isGood = (result && result === wpSwCache.urls[key]);
+                assert.isTrue(isGood);
+                if(isGood) {
+                  storageMatches++;
+                }
+              });
+            })*/;
+          })).then(function() {
+            assert.strictEqual(cacheMatches, keys.length);
+            done();
+          });
+        });
+      });
+    });
   });
 
-  it('A URL removed from the `urls` property removes unwanted URLs from cache and storage', function() {
+  it('A URL removed from the `urls` property removes unwanted URLs from cache and storage', function(done) {
     // This is key to removing stale files from cache after an admin no longer wants a file cached
     // Simulates a user caching files, going to admin to remove a file, and seeing that file removed from cache+storage
-    assert.strictEqual(true, true);
+    var removedUrl;
+
+    wpSwCache.update().then(function() {
+      removedUrl = Object.keys(wpSwCache.urls)[0];
+      delete wpSwCache.urls[removedUrl];
+
+      // The second "update" call simulates teh SW being re-installed/updated
+      wpSwCache.update().then(function() {
+        return wpSwCache.removeOldUrls();
+      }).then(function() {
+        // Ensure the URL is no longer in cache
+        return self.caches.open(wpSwCache.cacheName).then(function(cache) {
+          return cache.match(removedUrl).then(function(result) {
+            assert.strictEqual(result, undefined);
+          }).then(function() {
+            return wpSwCache.storage.getItem(removedUrl).then(function(hash) {
+              assert.strictEqual(hash, undefined);
```

Comment:
This storage check passes but I'm expecting `undefined` so I don't consider it 100% trustworthy until the issue in the `URLs which should be added to cache actually end up in cache and storage` test is figured out.